src/main/java/org/sagebionetworks/web/server/servlet/AppConfigServlet.java

```
@@ -0,0 +1,155 @@
+  public String configurationToken;
+  private StackConfiguration stackConfiguration;
+  private final String defaultConfigValue =
+    "{\"default configuration\":\"true\"}";
+
+  private String lastConfigValue = "";
+
+  private static final Logger logger = Logger.getLogger(
+    AppConfigServlet.class.getName()
+  );
+
+  @Inject
+  public AppConfigServlet(
+    AWSAppConfigData appConfigDataClient,
+    StackConfiguration stackConfiguration
+  ) {
+    this.appConfigDataClient = appConfigDataClient;
+    this.stackConfiguration = stackConfiguration;
+  }
+
+  @Override
+  public void init() throws ServletException {
+    super.init();
+    initializeAppConfigClient();
+  }
+
+  public void initializeAppConfigClient() {
+    if (appConfigDataClient == null) {
+      appConfigDataClient = AwsClientFactory.createAppConfigClient();
+    }
+    startConfigurationSession();
+    initializeConfigSupplier();
+  }
+
+  public void startConfigurationSession() {
+    try {
+      StartConfigurationSessionRequest sessionRequest =
+        new StartConfigurationSessionRequest()
+          .withApplicationIdentifier(
+            stackConfiguration.getStack() +
+            "-" +
+            stackConfiguration.getStackInstance() +
+            "-portal-AppConfigApp"
+          )
+          .withEnvironmentIdentifier(
+            stackConfiguration.getStack() +
+            "-" +
+            stackConfiguration.getStackInstance() +
+            "-portal-environment"
+          )
+          .withConfigurationProfileIdentifier(
+            stackConfiguration.getStack() +
+            "-" +
+            stackConfiguration.getStackInstance() +
+            "-portal-configurations"
+          );
+      StartConfigurationSessionResult sessionResponse =
+        appConfigDataClient.startConfigurationSession(sessionRequest);
+      configurationToken = sessionResponse.getInitialConfigurationToken();
+    } catch (Exception e) {
+      configurationToken = null;
+    }
```

Please log (or rethrow) the exception, a "swallowed" exception becomes incredibly hard to debug if something goes wrong.